### PR TITLE
[measurement-tools] expose a callback that allows the application to determine whether to show or hide the measurement tools.

### PR DESCRIPTION
**Pull request by thanhbinh-d**
The measuring tools have supported measurement on sheet views since PR [837](https://github.com/iTwin/viewer-components-react/pull/837). As a result, the changes made in PR [336](https://github.com/iTwin/viewer-components-react/pull/336), which disabled the measurement tools when a sheet view is active, need to be revised accordingly.

This PR introduces a callback that allows the application to determine whether to show or hide the measurement tools. Additionally, the tools now listen to the `SyncUiEventId.ActiveViewportChanged` event to update their visibility when the user changes the active view in a multi-view frontstage.

### CODE REVIEW

**Review by jason-crow — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

File: packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx [hunk 1]
```diff
@@ -107,8 +110,9 @@ export class MeasureToolsUiItemsProvider implements UiItemsProvider {
             MeasureToolDefinitions.clearMeasurementsToolCommand,
             {
               isHidden: new ConditionalBooleanValue(
-                () => isSheetViewActive() || !MeasurementUIEvents.isClearMeasurementButtonVisible,
+                () => !!this._props.isHidden?.(IModelApp.viewManager.selectedView) || !MeasurementUIEvents.isClearMeasurementButtonVisible,
```
[the remark this answers is not in the document]
Comment (Maxime-Brassard) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:113]: Sheet measurements aren't different than any other, they should all be clearable by the clear measurements button even while they aren't displayed (if, for example, they are not on the active sheet).
I believe this new logic (as well as the one that was present before) is meant to fix some bug where the clear measurement button would disappear when in a non-main view until the user clicks in the main view (at which point the button reappears and works as expected)

[the remark this answers is not in the document]
Comment (Maxime-Brassard) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:113]: There might be a better way to fix this problem though, like investigating the actual logic for this that directs the isClearMeasurementButtonVisible logic and find out why it behaves like that.

[the remark this answers is not in the document]
Comment (bsy-nicholasw) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:113]: OK so...`MeasurementUIEvents.isClearMeasurementButtonVisible=true`, but if you navigate to the sheet view, it becomes false?

Yeah, I would not expect an externally provided callback to tell a viewport is hidden or not to be an appropriate fix then if we have some problem internally.

[the remark this answers is not in the document]
Comment (Maxime-Brassard) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:113]: @thanhbinh-d Did what we write make sense so far with what you've seen?

[the remark this answers is not in the document]
Comment (thanhbinh-d) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:113]: **Background Information**
In PR 336, it was discussed that while some measurement tools work for drawings, they don't function for sheets. As a result, when a sheet view is active, the measurement tools (in the group button) and the clear measurements tool are hidden.

This PR addresses an issue where a measurement is added to a sheet view (which has the sheet transform data, as per PR 837), but the clear measurements tool remains invisible.

The behavior makes sense for single-view frontstages (like the 2D View in Pineapple), where users cannot select measurement tools, so there's no need to show the clear measurements tool.

However, in multi-view frontstages (like Civil), where multiple views can be open simultaneously, things become more complex. Here’s an example use case:
1. In the Civil frontstage, the measurement tools are visible in the vertical toolbar.
2. When a sheet view (without sheet transform data) is opened, the measurement tools become invisible, as expected.
3. After reactivating the 3D view, the measurement tools are visible again, as expected.
4. A user can select the Distance tool and start measuring on the 3D view. A measurement element is added, and the clear measurements tool is visible.
5. The user switches to the sheet view and uses the Distance tool. A measurement element is added to the sheet view (based on sheet coordinates), but the clear measurements tool becomes invisible.

This creates confusion, as users can see a measurement element on the sheet view, but the clear measurements tool is missing from the toolbar.

**My Opinion**
I agree with Nicholas' suggestion about the clear measurements tool visibility. We shouldn't base visibility on whether the sheet view is active or whether the sheet model has sheet transform data.

Instead, we should rely on a single condition: MeasurementUIEvents.isClearMeasurementButtonVisible being true or false. This way, if any measurement element is added (regardless of the active view), the clear measurements tool remains visible. This simplifies the logic and avoids potential confusion.

Does this approach make sense?

[the remark this answers is not in the document]
Comment (Maxime-Brassard) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx:113]: Makes perfect sense to me!

[the remark this answers is not in the document]
Comment (thanhbinh-d) [reply packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx]: @aruniverse, @jason-crow, @Maxime-Brassard, @bsy-nicholasw, I've just updated the code based on what we discussed above.
Thanks!

**Comment by raplemie** [comment]
@aruniverse Any idea what needs to be done so this E2E test works ? Seems that there is an issue with getting the credentials, does the user need to be part of iTwinjs org for these to go through ?

**Comment by aruniverse** [comment]
> @aruniverse Any idea what needs to be done so this E2E test works ? Seems that there is an issue with getting the credentials, does the user need to be part of iTwinjs org for these to go through ?

We are seeing e2e issues everywhere, we are actively working on fixing auth issues and will keep you posted.
Please see https://github.com/iTwin/auth-clients/issues/262

**Comment by thanhbinh-d** [comment]
@aruniverse , 
I am still not able to pass e2e test. I got the error "Failed OIDC signin" with "@itwin/oidc-signin-tool": "^4.3.6".
What should I do to overcome this issue?

Thanks,
Binh

**Comment by hl662** [comment]
> @aruniverse , I am still not able to pass e2e test. I got the error "Failed OIDC signin" with "@itwin/oidc-signin-tool": "^4.3.6". What should I do to overcome this issue?

Hi Binh, looking at the files changed as part of your PR request, you haven't bumped up the version of [oidc-signin-tool](https://github.com/iTwin/viewer-components-react/blob/318241bd863ed47e678e62efab6af9f96c354f20/apps/test-viewer/package.json#L36) to it's latest version yet, which would be [4.3.7](https://github.com/iTwin/auth-clients/blob/main/packages/oidc-signin-tool/package.json#L3). If you can update the number, then run pnpm install to update the lockfile, I expect the e2e tests in our CI pipelines to pass.

**Review by tcobbs-bentley — APPROVED** [review]

**Comment by thanhbinh-d** [comment]
@hl662 ,
I have the same error after bumping up oidc-signin-tool to 4.3.7.

**Comment by hl662** [comment]
> @hl662 , I have the same error after bumping up oidc-signin-tool to 4.3.7.

I'll investigate

**Comment by raplemie** [comment]
Closing to test if I can create a new one with my account.

**Comment by raplemie** [comment]
Reopening the original as my attempt did not work.

**Comment by anmolshres98** [comment]
After investigating this locally, the failure is a result of the test user and test environment mismatch. It seems that the test user being used is a qa test user and the sign in attempt is being made to a prod test client. The fix would be to edit the e2e test variable group so that the user and environments match. We are working on that now.

**Comment by grigasp** [comment]
> After investigating this locally, the failure is a result of the test user and test environment mismatch. It seems that the test user being used is a qa test user and the sign in attempt is being made to a prod test client. The fix would be to edit the e2e test variable group so that the user and environments match. We are working on that now.

This is interesting... Playwright is configured to start the webserver with QA parameters: https://github.com/iTwin/viewer-components-react/blob/17c9762d1ee9d126ba4c2fdf02cb2b155ba895bc/packages/itwin/tree-widget/playwright.config.ts#L80-L88. And variable group is configured to return QA test user. So everything should match. I think you should look at where the QA client variables get overriden to PROD ones.

**Comment by aruniverse** [comment]
Can you try recreating this pr not from a fork?

**Pull request closed without merging**